qemu/tests/boot_from_nbd_image.py

```diff
@@ -0,0 +1,57 @@
+import socket
+
+from virttest import qemu_storage
+
+from avocado import fail_on
+
+from avocado.utils import process
+
+from provider import qemu_img_utils as img_utils
+
+from provider.nbd_image_export import QemuNBDExportImage
+
+
+def run(test, params, env):
+    """
+    1) Create a local raw/qcow2/luks image
+    2) Export it with qemu-nbd
+    3) Convert remote system image to the exported nbd image
+    4) Start VM from the exported image
+    5) Log into VM
+    """
+    def _convert_image():
+        source = params['images'].split()[0]
+        target = params['convert_target']
+        source_params = params.object_params(source)
+        target_params = params.object_params(target)
+        source_image = qemu_storage.QemuImg(source_params, None, source)
+
+        # Convert source to target
+        fail_on((process.CmdError,))(source_image.convert)(
+            target_params, None, skip_target_creation=True)
+
+    nbd_export = QemuNBDExportImage(params["local_image_tag"], params)
+    nbd_export.create_image()
+    nbd_export.export_image()
+
+    # we only export image with local nbd server
+    localhost = socket.gethostname()
+    params['nbd_server_%s' % params['convert_target']
+           ] = localhost if localhost else 'localhost'
+
+    session = None
+    vm = None
+    try:
+        _convert_image()
+        vm = img_utils.boot_vm_with_images(test, params, env,
+                                           (params['convert_target'],))
+        session = vm.wait_for_login(
+            timeout=int(params.get("login_timeout", 360)))
+        if not session:
+            test.fail('Failed to log into VM')
```

Comment:
session could be None if booting vm failed, so here we did a check.

Reply:
For the codes of `wait_for_login`, I didn't found where routines will return `None`, could you show me you found?
And I test this functions:
```
>>> from virttest import remote
>>> session = remote.wait_for_login('ssh', '10.73.194.48', '22', 'root', 'kvmautotest', '^\[.*\][\#\$]\s*$', timeout=480)
Traceback (most recent call last):
  File "/home/yhong/for-bug-1847232/kar/workspace/avocado-vt/virttest/remote.py", line 160, in handle_prompts
    timeout=timeout, internal_timeout=0.5)
  File "/usr/local/lib/python3.6/site-packages/aexpect-1.5.1-py3.6.egg/aexpect/client.py", line 931, in read_until_last_line_matches
    print_func)
  File "/usr/local/lib/python3.6/site-packages/aexpect-1.5.1-py3.6.egg/aexpect/client.py", line 849, in read_until_output_matches
    raise ExpectTimeoutError(patterns, o)
aexpect.exceptions.ExpectTimeoutError: Timeout expired while looking for patterns ['[Aa]re you sure', '[Pp]assword:\\s*', '\\(or (press|type) Control-D to continue\\):\\s*$', '[Gg]ive.*[Ll]ogin:\\s*$', '(?<![Ll]ast )[Ll]ogin:\\s*$', '[Cc]onnection.*closed', '[Cc]onnection.*refused', '[Pp]lease wait', '[Ww]arning', '[Ee]nter.*username', '[Ee]nter.*password', '[Cc]onnection timed out', '^\\[.*\\][\\#\\$]\\s*$', 'Escape character is.*', 'Command>']    (output: '')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/yhong/for-bug-1847232/kar/workspace/avocado-vt/virttest/remote.py", line 428, in wait_for_login
    preferred_authenticaton=preferred_authenticaton)
  File "/home/yhong/for-bug-1847232/kar/workspace/avocado-vt/virttest/remote.py", line 306, in remote_login
    handle_prompts(session, username, password, prompt, timeout)
  File "/home/yhong/for-bug-1847232/kar/workspace/avocado-vt/virttest/remote.py", line 233, in handle_prompts
    raise LoginTimeoutError(e.output)
virttest.remote.LoginTimeoutError: Login timeout expired    (output: '')
```
It will raise `LoginTimeoutError` if fail to log into VM. not raise `None`.
PS: The above guest is a not bootable VM.

Reply:
hi @YongxueHong ,  something wrong.., my code change was gone...
update done, please have a double check, thanks much.